src/OpenTK.GraphicsLibraryFramework/Interfaces/IGLFW.cs

```diff
@@ -932,177 +1196,236 @@ internal interface IGLFW
+        /// </para>
         /// </summary>
         /// <param name="jid">The joystick to query.</param>
         /// <returns>
         /// <c>true</c> if a joystick is both present and has a gamepad mapping, or <c>false</c> otherwise.
         /// </returns>
         /// <remarks>
+        /// <para>
+        /// Call <see cref="JoystickPresent"/> to check if a joystick is present regardless of whether it has a mapping.
+        /// </para>
+        /// <para>
         /// This function must only be called from the main thread.
+        /// </para>
         ///
-        /// TODO: Possible errors include <see cref="ErrorCode.NotInitialized"/> and <see cref="ErrorCode.InvalidEnum"/>.
+        /// <para>
+        /// Possible errors include <see cref="ErrorCode.NotInitialized"/> and <see cref="ErrorCode.InvalidEnum"/>.
+        /// </para>
         /// </remarks>
         /// <seealso cref="GetGamepadState"/>
         bool JoystickIsGamepad(int jid);
 
         /// <summary>
+        /// <para>
         /// This function parses the specified ASCII encoded string
         /// and updates the internal list with any gamepad mappings it finds.
+        /// </para>
+        /// <para>
         /// This string may contain either a single gamepad mapping or many mappings separated by newlines.
+        /// </para>
+        /// <para>
         /// The parser supports the full format of the gamecontrollerdb.txt source file
         /// including empty lines and comments.
-        ///
+        /// </para>
+        /// <para>
         /// See <a href="https://www.glfw.org/docs/3.3/input_guide.html#gamepad_mapping">Gamepad mappings</a>
         /// for a description of the format.
-        ///
+        /// </para>
+        /// <para>
         /// If there is already a gamepad mapping for a given GUID in the internal list, it will be replaced by the one passed to this function. If the library is terminated and re-initialized the internal list will revert to the built-in default.
+        /// </para>
         /// </summary>
         /// <param name="newMapping">The string containing the gamepad mappings.</param>
         /// <returns><c>true</c> if successful, or <c>false</c> if an error occurred.</returns>
         /// <remarks>
+        /// <para>
         /// This function must only be called from the main thread.
-        ///
-        /// TODO: Possible errors include <see cref="ErrorCode.NotInitialized"/> and <see cref="ErrorCode.InvalidValue"/>.
+        /// </para>
+        /// <para>
+        /// Possible errors include <see cref="ErrorCode.NotInitialized"/> and <see cref="ErrorCode.InvalidValue"/>.
+        /// </para>
         /// </remarks>
-        /// <seealso cref="JoystickIsGamepad"/>
-        /// <seealso cref="GetGamepadName"/>
-        bool UpdateGamepadMappings(string newMapping); // TODO: use marshalling to ASCII C string
```

Comment:
here the marshalling question as well...

Reply:
In this case, the summary text says it has to be an ASCII string. If we could get some ADL-wizardry to marshal UTF strings to ASCII that'd be great, but the docs already mention the format it needs to be in.